internal/infrastructure/kubernetes/proxy/resource_provider_test.go

```diff
@@ -550,6 +544,15 @@ func TestDeployment(t *testing.T) {
 				})
 			}
 
+			if os.Getenv("UPDATE_TESTDATA") != "" {
```

Comment:
can we use this approach https://github.com/envoyproxy/gateway/blob/26d9d9794f0a01bc584bd0273fcfd902467115a0/internal/cmd/egctl/translate_test.go#L29

https://github.com/envoyproxy/gateway/blob/26d9d9794f0a01bc584bd0273fcfd902467115a0/tools/make/golang.mk#L49